[BUILD] Unable to build w/ MSYS2 Clang environment

From the build server log for the [CLANG64 environment](https://www.msys2.org/docs/environments/):

```
2022-02-16T00:37:56.6879764Z [3/181] Building CXX object src/libutil/CMakeFiles/OpenImageIO_Util.dir/benchmark.cpp.obj
2022-02-16T00:37:56.8399750Z FAILED: src/libutil/CMakeFiles/OpenImageIO_Util.dir/benchmark.cpp.obj 
2022-02-16T00:37:56.8783631Z D:\a\msys64\clang64\bin\clang++.exe -DOPENEXR_DLL -DOpenImageIO_EXPORTS -DOpenImageIO_Util_EXPORTS -DUSE_EXTERNAL_PUGIXML=1 -DUSE_FIELD3D=1 -DUSE_FREETYPE=1 -DUSE_JPEG_TURBO=1 -DUSE_OCIO=1 -DUSE_OPENCOLORIO=1 -DUSE_OPENCV=1 -DUSE_OPENVDB=1 -DUSE_STD_REGEX -DUSE_TBB=1 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -IC:/M/mingw-w64-openimageio/src/build-CLANG64/include/OpenImageIO -IC:/M/mingw-w64-openimageio/src/build-CLANG64/include -IC:/M/mingw-w64-openimageio/src/build-CLANG64/src/include -IC:/M/mingw-w64-openimageio/src/oiio-Release-2.2.18.0/src/include -I/clang64/include -isystem D:/a/msys64/clang64/include/OpenEXR -march=x86-64 -mtune=generic -O2 -pipe -O3 -DNDEBUG   -UUSE_OPENVDB -Wall -Wno-unused-function -Wno-overloaded-virtual -Wno-unneeded-internal-declaration -Wno-unused-private-field -Wno-tautological-compare -Qunused-arguments -Wunknown-warning-option -Wno-unused-local-typedefs -Wno-expansion-to-defined -fno-math-errno -Wno-deprecated-register -std=c++14 -MD -MT src/libutil/CMakeFiles/OpenImageIO_Util.dir/benchmark.cpp.obj -MF src\libutil\CMakeFiles\OpenImageIO_Util.dir\benchmark.cpp.obj.d -o src/libutil/CMakeFiles/OpenImageIO_Util.dir/benchmark.cpp.obj -c C:/M/mingw-w64-openimageio/src/oiio-Release-2.2.18.0/src/libutil/benchmark.cpp
2022-02-16T00:37:57.0330766Z In file included from C:/M/mingw-w64-openimageio/src/oiio-Release-2.2.18.0/src/libutil/benchmark.cpp:10:
2022-02-16T00:37:57.0702908Z C:/M/mingw-w64-openimageio/src/oiio-Release-2.2.18.0/src/include/OpenImageIO/thread.h:96:5: error: use of undeclared identifier 'sched_yield'
2022-02-16T00:37:57.2230704Z     sched_yield();
2022-02-16T00:37:57.2601670Z     ^
2022-02-16T00:37:57.4151223Z 1 error generated.
```

[Both g++ and clang++ define `__GNUC__`](https://github.com/OpenImageIO/oiio/blob/16e2bad47d2e75949ce640deed1013d15e25516d/src/include/OpenImageIO/thread.h#L90-L102), but only libstdc++ uses the pthreads model; libc++ uses the Win32 API on Windows so neither `pthread.h` nor `sched.h` are included implicitly by the C++ `<thread>` header.

I'm wondering if this should be removed completely and `std::this_thread_yield()` should simply be used [directly here](https://github.com/OpenImageIO/oiio/blob/16e2bad47d2e75949ce640deed1013d15e25516d/src/include/OpenImageIO/thread.h#L151) (both 2.2 and 2.3 branches)?

See also https://github.com/msys2/MINGW-packages/pull/10887 for a temporary/kludgy MSYS2 specific workaround (i.e. not applicable to Clang builds on other platforms).

--- a/src/include/OpenImageIO/thread.h
+++ b/src/include/OpenImageIO/thread.h
@@ -88,17 +88,11 @@ typedef std::lock_guard<recursive_mutex> recursive_lock_guard;
 
 
 /// Yield the processor for the rest of the timeslice.
-///
+/// DEPRECATED(2.4): Use std::this_thread::yield() instead.
 inline void
 yield() noexcept
 {
-#if defined(__GNUC__)
-    sched_yield();
-#elif defined(_MSC_VER)
-    SwitchToThread();
-#else
-#    error No yield on this platform.
-#endif
+    std::this_thread::yield();
 }
 
 
@@ -148,7 +142,7 @@ public:
             pause(m_count);
             m_count *= 2;
         } else {
-            yield();
+            std::this_thread::yield();
         }
     }
 

--- a/src/libutil/thread.cpp
+++ b/src/libutil/thread.cpp
@@ -499,7 +499,7 @@ task_set::wait_for_task(size_t taskindex, bool block)
         if (!m_pool->run_one_task(m_submitter_thread)) {
             // We tried to do a task ourselves, but there weren't any
             // left, so just wait for the rest to finish.
-            yield();
+            std::this_thread::yield();
         }
     }
 }
@@ -540,7 +540,7 @@ task_set::wait(bool block)
                 // We tried to do a task ourselves, but there weren't any
                 // left, so just wait for the rest to finish.
 #if 1
-                yield();
+                std::this_thread::yield();
 #else
                 // FIXME -- as currently written, if we see an empty queue
                 // but we're still waiting for the tasks in our set to end,

